src/main/kotlin/com/zepben/ewb/database/sqlite/cim/CimDatabaseReader.kt

```diff
@@ -59,14 +59,14 @@ abstract class CimDatabaseReader<TTables : CimDatabaseTables, TService : BaseSer
     /**
      * Read the database.
      */
-    fun read(service: TService): Boolean =
+    fun read(service: TService, afterServiceReadFunction: Boolean? = true): Boolean =
```

Comment:
suggestion: Unless a tri-state is required, it should just be a plain boolean rather than nullable. Also prefer boolean variables in particular to have names that are phrased as `is`, `has`, etc to give meaning to how it is used. I expected this to be a function to actually call after reading at first glance, thinking you had chosen the path of providing a nullable callback as the way of disabling this processing.

```suggestion
    fun read(service: TService, preformAfterReadProcessing: Boolean = true): Boolean =
```